packages/venia-concept/src/actions/cart/asyncActions.js

```diff
@@ -266,15 +264,14 @@ export const removeItemFromCart = payload => {
 
             // check if the guest cart has expired
             if (noGuestCartId || (response && response.status === 404)) {
-                // if so, then delete the cached ID...
-                // in contrast to the save, make sure storage deletion is
+                // Delete the cached ID from local storage.
+                // The reducer handles clearing out the bad ID from Redux.
+                // In contrast to the save, make sure storage deletion is
                 // complete before dispatching the error--you don't want an
                 // upstream action to try and reuse the known-bad ID.
                 await clearGuestCartId();
                 // then create a new one
                 await dispatch(createGuestCart());
-                // then retry this operation
-                return thunk(...arguments);
```

Comment:
No sense attempting to retry removing an item - we just got a brand new cart with nothing in it!